pkg/operator/discover/discover.go

```diff
@@ -44,8 +44,6 @@ import (
 const (
 	discoverDaemonsetName                 = "rook-discover"
 	discoverDaemonsetPriorityClassNameEnv = "DISCOVER_PRIORITY_CLASS_NAME"
-	discoverDaemonsetTolerationEnv        = "DISCOVER_TOLERATION"
```

Comment:
I wonder if we need to keep this for backward compatibility.

Reply:
@travisn I'm thinking to keep this value in go files and remove from yaml files. Also,  after reading from the env variable I'm thinking to update the configmap with the env values so that configMap has the values updated and from the next release, we could remove those completely and use the configmap as a source of truth.

thoughts?

Reply:
Yes, we can remove the env var from the examples, but for upgraded files we can still read it in the go code. We have had a pattern of only expecting the user to update the configmap, instead of doing it automatically. Since the helper method will fall back to the env var if the configmap doesn't have the value, it's a simple behavior to maintain.

Reply:
make sense, done